src/rt/sched/schedulerlist.h

```diff
@@ -0,0 +1,131 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include "ds/dllist.h"
+#include "threadsync.h"
+
+#include <cstdlib>
+#include <snmalloc/snmalloc.h>
+
+namespace verona::rt
+{
+  // SchedulerList implements two lists, active and free, both protected by the
+  // same lock. Implementing this here rather than inside the ThreadPool
+  // decouples its logic from the rest of the runtime implementation and should
+  // allow to easily change this class.
+  template<class T>
+  class SchedulerList
+  {
+  public:
+    snmalloc::FlagWord m;
+
+    DLList<T> active;
+    DLList<T> free;
+
+  public:
+    SchedulerList() {}
+    ~SchedulerList()
+    {
+      snmalloc::FlagLock lock(m);
```

Comment:
I like having debug information here, but it might prevent the inlining of the structure into the thread pool.  As this is not perf sensitive, then it might be fine to leave it as a separate allocation.